src/scheduler/drr.rs

```diff
@@ -155,4 +162,166 @@ impl<const HASH_PORTS: bool> Scheduler for Drr<HASH_PORTS> {
         self.limit_bytes = bytes;
         Ok(())
     }
+    fn dbg(&mut self, _epoch_dur: Duration) {
+        self.log()
+        
+    }
+
+    
 }
+
+
+impl<const HASH_PORTS: bool, L: std::io::Write> Drr<HASH_PORTS, L> {
+    pub fn with_logger(self, w : L) -> Drr<HASH_PORTS,L> {
+        self.maybe_with_logger(Some(w))
+    }
+
+    pub fn maybe_with_logger(self, w: Option<L>) -> Drr<HASH_PORTS, L> {
+        Drr {
+            limit_bytes: self.limit_bytes,
+            queues: self.queues,
+            curr_qsizes: self.curr_qsizes,
+            deficits: self.deficits,
+            quanta: self.quanta,
+            queue_map: self.queue_map,
+            num_queues: self.num_queues,
+            deq_curr_qid: self.deq_curr_qid,
+
+            logger: w.map(|x| csv::WriterBuilder::new().has_headers(false).from_writer(x)),
+        }
+        
+    }
+    pub fn log(&mut self){
+        
+        if let Some(log) = self.logger.as_mut() {
+           /*  struct Flow <'a>{
+                protocol: u8,
+                source_ip: &'a str,
+                dest_ip: &'a str,
+                sport: u16,
+                dport: u16
+            }
+            impl Serialize for Flow<'a> {
+                fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+                where
+                    S: Serializer,
+                {
+                    // 3 is the number of fields in the struct.
+                    let mut state = serializer.serialize_struct("Flow", 5)?;
+                    state.serialize_field("protocol", &self.protocol)?;
+                    state.serialize_field("source_ip", &self.source_ip)?;
+                    state.serialize_field("dest_ip", &self.dest_ip)?;
+                    state.serialize_field("sport", &self.sport)?;
+                    state.serialize_field("dport", &self.dport)?;
+                    state.end()
+                }
+            }*/
+                        
+            #[derive(serde::Serialize)]
+            struct Record{
+                unix_time_ms: u128,
+                queue_id: usize,
+                queue_size: usize,
+                flows_protocols: Vec<u8>,
+                flows_source_ip: Vec<String>,
+                flows_dest_ip: Vec<String>,
+                flows_sport: Vec<u16>,
+                flows_dport: Vec<u16>,
```

Comment:
If we're going to `format!` something anyway, it would be more readable to print in a format like `tcp:1.1.1.1:2222->3.3.3.3:4444`.

Basically, if you were reading this log, what format would you want to read it in?